capstone-frontend/src/pages/UserProfile.jsx

```diff
@@ -26,7 +28,7 @@ function UserProfile() {
   };
   let userIndex = 0;
   for (const item in userData) {
-    if (userData[item].googleId == userid) {
+    if (userData[item].email == userid) {
```

Comment:
If you just want a specific user here, then your backend should just return that specific user (again based on cookie or URL param). This could be a separate endpoint.

You may not even need to do this fetch, if you use cookie below for the userposts endpoint.

Reply:
If you *do* need this, you should fetch it at the root (App.jsx) and put it in context or similar.

Reply:
Transitioned into using Session Storage to transfer user information; will attempt to do so using this.